File: data/src/main/java/com/github/mrmitew/skeleton/data/http/BackendRestApi.java
package com.github.mrmitew.skeleton.data.http;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface BackendRestApi {
    @GET("users/{user}/repos")
    Single<List<Object>> listRepos(@Path("user") String user);
}

File: data/src/main/java/com/github/mrmitew/skeleton/data/http/BackendRestApi.kt
package com.github.mrmitew.skeleton.data.http

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface BackendRestApi {

    @GET("users/{user}/repos")
    fun listRepos(@Path("user") user:String) : Single<List<Any>>

}